### PR TITLE
Add support to the CLI command `run` for converting a config from a different format

### LINKED ISSUE #3737: Wire up the Convert API to the `fmt` command for translating supported configs to river

### LINKED ISSUE #3738: Wire up the Convert API to the `run` command for translating supported configs to river on the fly

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### Which issue(s) this PR fixes

Closes #3737
Closes #3738

#### Notes to the Reviewer

We should begin work on https://github.com/grafana/agent/issues/4073 right after this and complete it before release to take a wider view of docs for the converter feature.

Testing notes: I ran a prometheus config and confirmed it was scraping/forwarding metrics to grafana cloud using this. I also reloaded  with the API after updating the config.yaml and saw the update in the debug UI. I tested with an unsupported feature and confirmed the converter.bypass-warnings flag must be provided or the agent won't start up.

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: cmd/internal/flowmode/cmd_run.go
Comment (rfratto) [inline cmd/internal/flowmode/cmd_run.go:105]: What do you think about putting these flags in the `config` namespace to make it more clear what's happening, e.g., `--config.format` and `--config.bypass-conversion-warnings`? 

`converter` makes it sound like there's some converter process always running, when really this is just processing done on the config file. 

(When I was playing around with the `convert` subcommand, I found `--source-format` to be weirdly long; `--format` might make enough sense since the target format is always Flow)

File: docs/sources/flow/reference/cli/run.md
Comment (rfratto) [inline docs/sources/flow/reference/cli/run.md:79]: "Mode" tends to be reserved for operational modes (static mode, flow mode), so I don't want to overload the term here. 

Can we just call it what it is? Something like "Configuration conversion?"

File: cmd/internal/flowmode/cmd_run.go
Comment (erikbaranowski) [reply cmd/internal/flowmode/cmd_run.go]: done

File: docs/sources/flow/reference/cli/run.md
Comment (erikbaranowski) [reply docs/sources/flow/reference/cli/run.md]: done

**Review by rfratto — APPROVED** [review]
LGTM pending documentation review from Clayton. 

Thinking about this, though, I have a question: should we be calling this out as a beta feature? As it is right now, it's defaulting to stable, but I don't know if we're ready to call it that yet.

(It's also not experimental because I think we're set on supporting this use case)

**Comment by erikbaranowski** [comment]
Good call on calling it beta for at least 1 release. I will addd that to the header here and in the convert command doc tomorrow for this PR

File: CHANGELOG.md
Comment (thampiotr) [inline CHANGELOG.md:36]: duplicated line

File: cmd/internal/flowmode/cmd_run.go
Comment (thampiotr) [inline cmd/internal/flowmode/cmd_run.go:104]: Isn't default config format `river`? Should we make it the default and a supported option? I think this makes sense for a flag called `config.format`.

Alternatively, if we want this flag to mean "convert config from this format to river", then we could name the flag differently, e.g. `config.convert-format`".

File: CHANGELOG.md
Comment (erikbaranowski) [reply CHANGELOG.md:36]: This includes a wording fix for a previous PR on `grafana-agent convert` that was originally `grafana-agent fmt`.

File: docs/sources/flow/reference/cli/run.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/cli/run.md:91]:
```suggestion
If you also use the `--config.bypass-conversion-warnings` command-line argument,
Grafana Agent will ignore any warnings from the converter. Use this
argument with caution because the resulting conversion is not equivalent to the
original configuration.
```

**Review by clayton-cornell — APPROVED** [review]
Some small suggestions to shift the doc input to more active voice.

File: cmd/internal/flowmode/cmd_run.go
Comment (erikbaranowski) [reply cmd/internal/flowmode/cmd_run.go:104]: +1 to `config.convert-format`. @rfratto do we have consensus on this?

Comment (rfratto) [reply cmd/internal/flowmode/cmd_run.go:104]: `-config.format` defaulting to `river` (ie apply no conversion) is a good idea. 

I'm not sure about `config.convert-format`, to me it implies converting TO a format, rather than converting FROM a format. 

I don't want to block on bikeshedding though, final call is yours.

Comment (erikbaranowski) [reply cmd/internal/flowmode/cmd_run.go:104]: I can update the default and code it to handle `river` as existing behavior.

Comment (rfratto) [reply cmd/internal/flowmode/cmd_run.go:104]: > I can update the default and code it to handle river as existing behavior.

I don't think that's necessary, but the CLI could understand river as "no op", and not-river as "do a conversion"


BTW: I don't think it should be `river`, but `flow` instead; River is the language, Flow is the expected schema of a config file written in that language. It's the same reason one of the expected formats is Prometheus (the schema) and not YAML (the language)

Comment (erikbaranowski) [reply cmd/internal/flowmode/cmd_run.go:104]: makes sense. I have to add 1 small conditional to not call the converter if `flow` otherwise it will pass it along.

Comment (erikbaranowski) [reply cmd/internal/flowmode/cmd_run.go]: This change has been implemented. I'll merge this tomorrow pending any last minute feedback.

Comment (rfratto) [reply cmd/internal/flowmode/cmd_run.go]: LGTM 👍

**Review by thampiotr — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
